products/monitoring/api.yaml

```diff
@@ -964,6 +962,141 @@ objects:
             Formatted as described in
             https://cloud.google.com/apis/design/resource_names.
 
+  - !ruby/object:Api::Resource
+    name: Slo
+    base_url: projects/{{project}}/services/{{service}}/serviceLevelObjectives
+    # name = projects/{{project}}/services/{{service}}/serviceLevelObjectives/{{slo_id}}
+    self_link: "{{name}}"
+    create_url: projects/{{project}}/services/{{service}}/serviceLevelObjectives?serviceLevelObjectiveId={{slo_id}}
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A Service-Level Objective (SLO) describes the level of desired good
+      service. It consists of a service-level indicator (SLI), a performance
+      goal, and a period over which the objective is to be evaluated against
+      that goal. The SLO can use SLIs defined in a number of different manners.
+      Typical SLOs might include "99% of requests in each rolling week have
+      latency below 200 milliseconds" or "99.5% of requests in each calendar
+      month return successfully."
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Service Monitoring': 'https://cloud.google.com/monitoring/service-monitoring'
+        'Monitoring API Documentation': 'https://cloud.google.com/monitoring/api/v3/'
+      api: 'https://cloud.google.com/monitoring/api/ref_v3/rest/v3/services.serviceLevelObjectives'
+    parameters:
+    - !ruby/object:Api::Type::String
+      name: service
+      required: true
+      url_param_only: true
+      input: true
+      description: |
+        ID of the service to which this SLO belongs.
+    - !ruby/object:Api::Type::String
+      name: sloId
+      description: |
+        The id to use for this ServiceLevelObjective. If omitted, an id will be generated instead.
+      input: true
+    properties:
+    - !ruby/object:Api::Type::String
+      name: name
+      description: |
+        The full resource name for this service. The syntax is:
+        projects/[PROJECT_ID_OR_NUMBER]/services/[SERVICE_ID]/serviceLevelObjectives/[SLO_NAME]
+      output: true
+    - !ruby/object:Api::Type::String
+      name: displayName
+      description: |
+        Name used for UI elements listing this SLO.
+    - !ruby/object:Api::Type::Double
+      name: goal
+      required: true
+      description: |
+        The fraction of service that must be good in order for this objective
+        to be met. 0 < goal <= 0.999
+    - !ruby/object:Api::Type::String
+      name: rollingPeriod
+      exactly_one_of:
+        - rolling_period
+        - calendar_period
+      description: |
+        A rolling time period, semantically "in the past <rollingPeriod>".
+        Must be an integer multiple of 1 day no larger than 30 days.
```

Comment:
I'm not totally sure from reading this how a user would specify it. Is this an integer, or a string?

Reply:
it is actually very dumb and must be X days in SECONDS (e.g. "rolling_period = 86400s") - I was wondering whether we wanted to change this to be an number and encode/decode this value (there are a couple of them)

Reply:
This is a really weird field and something i wanted to get commentary on (same with basicSli.latency.threshold) - based on my attempts to set this value, I believe the monitoring API only allows for "Xs", which means this needs to be like "$int days" in seconds, e.g. "86400s". Should we make this an int field for number of days and convert to the API-expected format?

Reply:
I'd worry a bit about whether they eventually remove that restriction, though if we rename the field to "rolling_period_days" then we could go back and add a more generic "rolling_period" later. Maybe bring it up with the stackdriver folks to see if they have additional insight? What does gcloud do?

I think that if it ends up eventually supporting other types of duration strings then we'd want to leave it as-is, but if it's going to be seconds or days forever then an int would make more sense.

Reply:
Talked to them, and they are ok with having it as days (they originally wanted to have it as an int for days) - i'll add a rolling_period_days field

Reply:
Looks good! quick nit: maybe remove the part about "must be an integer multiple of 1 day" since that's built in now.